repo/incubating/elastic/versions/0/elastic-frameworkversion.yaml

```diff
@@ -0,0 +1,370 @@
+                    value: {{NAME}}-cluster
+                  - name: discovery.seed_hosts
+                    value: {{NAME}}-master-0.{{NAME}}-master-hs,{{NAME}}-master-1.{{NAME}}-master-hs,{{NAME}}-master-2.{{NAME}}-master-hs
+                  - name: node.master
+                    value: "false"
+                  - name: node.data
+                    value: "false"
+                  - name: node.ingest
+                    value: "false"
+                  - name: cluster.remote.connect
+                    value: "false"
+                volumeMounts:
+                  - name: data
+                    mountPath: /usr/share/elasticsearch/data
+        volumeClaimTemplates:
+          - metadata:
+              name: data
+            spec:
+              accessModes: [ "ReadWriteOnce" ]
+              resources:
+                requests:
+                  storage: 1Gi
+    ingest-service.yaml: |
+      kind: Service
+      apiVersion: v1
+      metadata:
+        name: ingest-hs
+        namespace: {{NAMESPACE}}
+      spec:
+        selector:
+          app: ingest
+        ports:
+          - protocol: TCP
+            port: 9200
+        clusterIP: None
+    ingest.yaml: |
+      kind: StatefulSet
+      apiVersion: apps/v1
+      metadata:
+        name: ingest
+        namespace: {{NAMESPACE}}
+      spec:
+        selector:
+          matchLabels:
+            app: ingest # has to match .spec.template.metadata.labels
+        serviceName: {{NAME}}-ingest-hs
+        replicas: {{INGEST_NODE_COUNT}}
+        template:
+          metadata:
+            labels:
+              app: ingest # has to match .spec.selector.matchLabels
+          spec:
+            initContainers:
+              - name: init-sysctl
+                image: busybox
+                command: ['sh', '-c', 'sysctl -w vm.max_map_count=262144']
+                securityContext:
+                  privileged: true
+              - name: volume-permissions
+                image: busybox
+                command: ['sh', '-c', 'chown -R 1000:1000 /usr/share/elasticsearch/data']
+                volumeMounts:
+                  - name: data
+                    mountPath: /usr/share/elasticsearch/data
+            terminationGracePeriodSeconds: 10
+            containers:
+              - name: elastic
+                image: elasticsearch:7.0.0
+                ports:
+                  - containerPort: 9200
+                    name: api
+                  - containerPort: 9300
+                    name: internal
+                env:
+                  - name: cluster.name
+                    value: {{NAME}}-cluster
+                  - name: discovery.seed_hosts
+                    value: {{NAME}}-master-0.{{NAME}}-master-hs,{{NAME}}-master-1.{{NAME}}-master-hs,{{NAME}}-master-2.{{NAME}}-master-hs
+                  - name: node.master
+                    value: "false"
+                  - name: node.data
+                    value: "false"
```

Comment:
node.master, node.data, node.ingest configure the personality of the es node in this case we configure an ingest node, this can not be changed by a user.

cluster.remote.connect is for cross cluster search and cross cluster replication, we not getting into this at this point.